File: corpus/go/fixture_24.go
package fixture24

type PriceBook24 struct {
    Multiplier float64
}

func CalculateDiscount24(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 240.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}

File: corpus/typescript/fixture_24.ts
type CartItem24 = { price: number; qty?: number };
export function calculateDiscount24(cartItems: CartItem24[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 240 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook24 {
  lookupPrice(sku: string): number { return sku.length + 24; }
}
